### ISSUE TITLE
multicursor: operator-modifier (like Vim Mode Plus for Atom)

### ISSUE THREAD

**Issue by mbahnmiller** [neovim/neovim#21334]
### Feature description

Here is an in insanely useful feature - described here:
https://github.com/t9md/atom-vim-mode-plus/wiki/AdvancedTopicTutorial#5-modify-operators-behavior-by-o-to-make-it-operate-on-occurrence

I don't think I can give a description of it justice, but it's extremely handy to be able to mark "occurrences" of a thing (by doing a visual select and type "g" "o"), then perform an action, such as "c" "i" "p", which would normally change inner paragraph, but having a pre-marked "occurrences" will automatically put you in multi-cursor mode on all of the selected occurrences within the target.

Additionally, instead of pre-marking the occurrences, performing an action like "c" "o" "i" "p" would automatically use the token which the cursor is currently located on as the target.

Example

This is the text.
I like the text here.
Here is more text, and this is also the text.

If your cursor were on any of the "text" occurrences, and you typed coipbanana<esc>, then all occurrences of "text" will now be changed with "banana":

This is the banana.
I like the banana here.
Here is more banana, and this is also the banana.

This really simplifies multiple cursors when you already know what your target and "occurrence" (or token or whatever you want to call it) are.

**Comment by Spirarel** [comment]
YES! This one feature made Atom more ergonomic for me than neovim. It's so absurdly useful.

**Comment by IndianBoy42** [comment]
Full multicursor would be nice but for now, for `co...` you can use gbprod/substitute.nvim. `vim.keymap.set("n", "co", require('substitute.range').word, { noremap = true })` gets you basically there. It uses the inbuilt `:s` command and its incremental preview so it looks pretty smooth.
